### DISCUSSION TITLE
Question about running a 3D event loop in background

### DISCUSSION THREAD

**Discussion by tamasgal** [JuliaPlots/Makie.jl#4253] [Q&A]
I would like to allow the user to interact with the 3D event loop and I could not figure out yet how. I tried to put the event loop in background with `@async` but the main thread is very unresponsive.

Currently I have something like this as my event loop, which reacts to changes in `simparams` and other stuff:

```julia
function eventloop(scene::Scene)
    while isopen(screen)
        if simparams.quit
            simparams.quit = false
            break
        end

        # display lots of stuff, depending on some settings from `simparams`, which is a global settings struct 

        GLMakie.pollevents(screen)
        GLMakie.render_frame(screen)

        GLFW.SwapBuffers(GLMakie.to_native(screen))

        if !isstopped()
            simparams.frame_idx += simparams.speed
        end
    end
end
```

and this is what I would like to achieve, demonstrated in a dummy session. Just to clarify, the code until `init_and_start_eventloop()` works fine and I can call `faster(10)` and other mutators via keybindings on the event handler, so technically everything is there. What's missing is **giving back the REPL to the user** after creating the event loop.

```
julia> @async init_and_start_eventloop()  # runs some code and launches the `eventloop` above

julia> faster(10)  # modifies some state and the 3D event loop will adjust accordingly

julia> hide_mc_tracks()

julia> shutdown()  # calls  GLMakie.destroy!(screen)
```

Can someone toss me in the right direction?

Edit: forgot to mention that I also tried things like `Threads.@spawn eventloop(scene)` and alike, but it only hangs with a grey screen.

**Comment by ffreyer** [comment]
Bit late now, but you probably need to yield or sleep to give other tasks an opportunity to run.

In general though you should just leave this to Makie. If the reason you created your own renderloop was to have something happen once per frame you can now use `events(fig).tick` for that.
